bindings/pyroot/pythonizations/python/ROOT/pythonization/_roofit/__init__.py

```diff
@@ -30,8 +30,33 @@
     python_classes_dict[python_class.__name__] = python_class
 
 
-def ismagicfunc(name):
-    return name.startswith("__") and name.endswith("__")
+def get_defined_attributes(klass):
+    """
+    Get all class attributes that are defined in a given class or in any of its
+    base classes (except for `object`).
+    """
+
+    blacklist = ["__dict__", "__doc__", "__hash__", "__module__", "__weakref__"]
+
+    # get a list of this class and all its base classes, excluding `object`
+    method_resolution_order = klass.mro()
```

Comment:
Why is it necessary that you also take the methods of base classes? Shouldn't you just inject in the Python proxy class the methods that were defined in your mirror class (and strictly just those methods)?

Reply:
I also want to get pythonizations from the base classes of the mirror class.

The point of mirroring also the inheritance hierachy was to inherit pythonizations even if the function gets redefined on the C++ side. In this case, cppyy doesn't pythonize the redefinition (which is correct in general), but often want to use the same pythoniztion by default (for example for `RooAbsReal::plotOn` and `RooAbsPdf::plotOn`).

Reply:
I see. In that case, I would just redefine the pythonization too in the derived class mirror class (you can invoke some helper function that is shared for base and derived, to avoid code duplication). I think this is cleaner: pythonizations are automatically inherited if there is no redefinition of the C++ side; if there is a redefinition on the C++ side, the pythonization is redefined too, to explicitly state that this is your desire (otherwise you are implicitly taking the same decision for every case like this). Moreover, this will allow you to get rid of walking on the methods of the base classes.

Reply:
Okay, we can do it like this!